src/tribler/core/start_core.py

```diff
@@ -179,7 +183,7 @@ def run_tribler_core_session(api_port: int, api_key: str, state_dir: Path, gui_t
     return exit_code
 
 
-def run_core(api_port, api_key, root_state_dir, parsed_args):
+def run_core(api_port: Optional[int], api_key: Optional[str], root_state_dir, parsed_args):
```

Comment:
Here, `api_key` does not feel optional. If it is None, the process is killed.

Reply:
It is an interesting topic.
* On one side, I agree that the `None` value always leads to a process termination with an exit code
* But I want to save the error code and message to the process state in the process database
* Currently, the `run_core` function we discuss here creates an object that represents the process instance. For the Core process, the sequence is the following:
    1. First, the `run_tribler.py` script started. It is a simple script that reads process arguments and environment variables and then runs either the `run_gui` function or the `run_core` function.
    2. `run_core` function creates the `current_process` object and checks that the core process can be started (that is, it checks that there is no other core process already running and that core-related arguments are correct).
    3. If the core process should and can be started, `run_tribler_core_session` is called that run a session with all core components.
    
    That is, `run_core` is an intermediate layer between the `run_tribler` script and the `run_tribler_core_session` function that handles conditions on whether the core process can be started. It accepts `None` as a possible `api_key` vale and correctly stops the current process with an appropriate error message.
    
We can make the `api_key` argument non-optional, but then we should handle the error outside the `run_core` function, that is, directly in the `run_tribler` script. In this case, we need to instantiate the `current_process` object right inside the `run_tribler` script, and in this case, the `run_tribler` will not be a simple script anymore that just read arguments and environment variables and calls either `run_gui` or `run_core`. The logic of the `run_tribler` script will be much more complicated in this case.

For that reason, I think it is a reasonable approach to allow the intermediate `run_core` function to handle the case of missing the `api_key` value, and in that case, the argument should be optional.

Reply:
In the current main, core still works if the api_key is None so I would like to keep that behavior intact. Since this PR is targeted towards allowing a free port, I would consider changes on api_key behavior out of scope and would warrant its own PR and separate discussion on whether or not to change the default behavior.

Reply:
I agree. I removed the check for `api_key`, and now it is possible to run core without specifying a CORE_API_KEY value again.